chaossearch/resource_object_group.go

```diff
@@ -98,9 +106,20 @@ func resourceObjectGroup() *schema.Resource {
 	}
 }
 
+
+
 func resourceObjectGroupCreate(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	c := meta.(*ProviderMeta).Client
 
+	// "unlimited" flattening represented as "null" in the api, and as -1 in the terraform module
+	// because the terraform sdk doesn't support nil values in configs https://github.com/hashicorp/terraform-plugin-sdk/issues/261
+	// We represent "null" as an int pointer to nil in the code.
+	array_flatten_tf := data.Get("array_flatten_depth").(int)
+	var array_flatten_cs *int = nil
+	if (array_flatten_tf != -1) {
+		array_flatten_cs = &array_flatten_tf
```

Comment:
will it work if you remove line 118 and just do `array_flatten_cs := &array_flatten_tf` here?